feature/gribi/mpls_in_udp/otg_tests/mpls_in_udp_scale/mpls_in_udp_scale_test.go

```diff
@@ -0,0 +1,1757 @@
+	} else if mplsPacketCount == 0 {
+		t.Errorf("No UDP-IPv6 packets found in capture on port %s", otgPortName)
+	} else if validMplsPacketCount == 0 {
+		t.Errorf("No valid MPLS-in-UDP packets found in capture on port %s", otgPortName)
+	} else if validMplsPacketCount < (mplsPacketCount / 2) {
+		t.Errorf("Many packets (%d/%d) failed validation", mplsPacketCount-validMplsPacketCount, mplsPacketCount)
+	} else {
+		t.Logf("Packet capture validation PASSED: Found %d valid MPLS-in-UDP packets", validMplsPacketCount)
+	}
+}
+
+// Main test entry point.
+func TestMPLSinUDPScale(t *testing.T) {
+	t.Helper()
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+	ateConfig := gosnappi.NewConfig()
+	ctx := context.Background()
+	vrfsList := configureDUT(t, dut)
+	configureOTG(t, ate, dut, ateConfig)
+	ate.OTG().PushConfig(t, ateConfig)
+	ate.OTG().StartProtocols(t)
+	otgutils.WaitForARP(t, ate.OTG(), ateConfig, "IPv6")
+
+	// Configure gRIBI client
+	c := gribi.Client{
+		DUT:         dut,
+		FIBACK:      true,
+		Persistence: true,
+	}
+
+	if err := c.Start(t); err != nil {
+		t.Fatalf("gRIBI Connection can not be established")
+	}
+	defer c.Close(t)
+	c.BecomeLeader(t)
+
+	// Flush all existing AFT entries and set up basic routing infrastructure
+	c.FlushAll(t)
+	programBasicEntries(t, dut, &c)
+	t.Run("Profile-1-Single VRF", func(t *testing.T) {
+		configureVrfProfiles(t, ateConfig, ctx, dut, ate, &c, []string{deviations.DefaultNetworkInstance(dut)}, 1)
+	})
+	t.Run("Profile-2-Multi VRF", func(t *testing.T) {
+		configureVrfProfiles(t, ateConfig, ctx, dut, ate, &c, vrfsList, 2)
+	})
+	t.Run("Profile-3-Multi VRF with Skew", func(t *testing.T) {
+		configureVrfProfiles(t, ateConfig, ctx, dut, ate, &c, vrfsList, 3)
+	})
+	t.Run("Profile-4-Single VRF", func(t *testing.T) {
+		configureVrfProfiles(t, ateConfig, ctx, dut, ate, &c, []string{deviations.DefaultNetworkInstance(dut)}, 4)
+	})
+	t.Run("Profile-5-Single VRF", func(t *testing.T) {
+		configureVrfProfiles(t, ateConfig, ctx, dut, ate, &c, []string{deviations.DefaultNetworkInstance(dut)}, 5)
+	})
+}
+
+// configureVrfProfiles implements the “Single/Multi VRF Validation” for Profile 1 (baseline) and Profile 4 (ECMP).
+// It programs MPLS-in-UDP NHs, NHGs, and 20k prefixes (10k v4 + 10k v6),
+// validates FIB/AFT, sends traffic, checks MPLS-over-UDP encapsulation, and deletes entries.
+func configureVrfProfiles(
+	t *testing.T,
+	ateConfig gosnappi.Config,
+	ctx context.Context,
+	dut *ondatra.DUTDevice,
+	ate *ondatra.ATEDevice,
+	c *gribi.Client,
+	vrfs []string,
+	profile int, // 1 = baseline, 4 = ECMP
+) {
+	t.Helper()
+	cfg, ok := profiles[profile]
+	if !ok {
+		t.Fatalf("Unsupported profile %d", profile)
+	}
+	totalNHGs := cfg.totalNHGs
+	nhsPerNHG := cfg.nhsPerNHG
+	totalNHs := totalNHGs * nhsPerNHG
+	var entries []fluent.GRIBIEntry
+	var wantAdds []*client.OpResult
+	var wantDels []*client.OpResult
+	var flows []gosnappi.Flow
+	var labelList []uint64
```

Comment:
Use a common `var()` block.


```suggestion
	var (
		entries []fluent.GRIBIEntry
		wantAdds []*client.OpResult
		wantDels []*client.OpResult
		flows []gosnappi.Flow
		labelList []uint64
	)
```